calico/felix/frules.py

```diff
@@ -163,27 +167,21 @@ def install_global_rules(config, v4_filter_updater, v6_filter_updater,
     # then this string is "tap+".
     iface_match = config.IFACE_PREFIX + "+"
 
-    iptables_generator = config.plugins["iptables_generator"]
-
     # If enabled, create the IP-in-IP device
     if config.IP_IN_IP_ENABLED:
         _log.info("IP-in-IP enabled, ensuring device exists.")
-        if not devices.interface_exists(IP_IN_IP_DEV_NAME):
-            # Make sure the IP-in-IP device exists; since we use the global
-            # device, this command actually creates it as a side-effect of
-            # initialising the kernel module rather than explicitly creating
-            # it.
-            _log.info("Tunnel device didn't exist; creating.")
-            futils.check_call(["ip", "tunnel", "add", IP_IN_IP_DEV_NAME,
-                               "mode", "ipip"])
-        futils.check_call(["ip", "link", "set", IP_IN_IP_DEV_NAME, "mtu",
-                           str(config.IP_IN_IP_MTU)])
-        if not devices.interface_up(IP_IN_IP_DEV_NAME):
-            _log.info("Tunnel device wasn't up; enabling.")
-            futils.check_call(["ip", "link", "set", IP_IN_IP_DEV_NAME, "up"])
+        try:
+            _configure_ipip_device(config)
+        except FailedSystemCall:
```

Comment:
It only retries once.  An error from the retry will take down felix and we'll let init restart us.